### ISSUE TITLE
@typescript-eslint/semi function declarations should have semi at end, throws error

### ISSUE THREAD

**Issue by ctfrancia** [typescript-eslint/typescript-eslint#971]
**Repro**

```javascript
module.exports =  {
  parser:  '@typescript-eslint/parser', 
  extends:  [
    'plugin:@typescript-eslint/recommended',
    'plugin:@typescript-eslint/eslint-recommended',
    'prettier/@typescript-eslint',
    'plugin:prettier/recommended',
  ],
 parserOptions:  {
    ecmaVersion:  2018, 
    sourceType:  'module',
  },
  rules:  {
    "semi": "off",
    "@typescript-eslint/semi": ["error"]
  },
};
```

```TS
public someFunc = (input: string): string | null => {
  //some code
}; // <--- ERROR: Unnecessary semicolon
```

**Expected Result**
I would like to have it recognize that it's a definition and therefore a semicolon is ok

**Actual Result**
error as stated above. Whenever I try to remove the semi colon then there  is a prettier issue and when I save the semi colon is placed back, which makes sense.
```typescript
const obj = {
  foo: 'bar',
}; ✅
```
```typescript
const func = () => {
 return true;
}; ❗️
```
**Additional Info**
[this is a cross post](https://stackoverflow.com/questions/57903821/eslint-function-definitions-should-require-semi-colon-but-has-error)

I would like to note that I have tried many different possibilities with rules, adding, removing, etc.
tried following [this](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/semi.md) section to disable it/allow a semi colon at the end

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.13.0` |
| `@typescript-eslint/parser`        | `1.13.0` |
| `TypeScript`                       | `3.6.3` |
| `ESLint`                           | `6.3.0` |
| `node`                             | `12.7.0` |
| `npm`                              | `6.10.2` |
